lib/Onyx.js

```diff
@@ -780,6 +780,39 @@ function mergeCollection(collectionKey, collection) {
         });
 }
 
+/**
+ * Insert API responses and lifecycle data into Onyx
+ *
+ * @param {Object} data An object containing the Onyx data to set
+ * @param {String} data.onyxMethod One of [set, merge] Onyx methods
+ * @param {String} data.key ONYXKEY to set
+ * @param {*} data.value Value to store in ONYXKEY
+ */
+function update(data) {
+    // First, validate the Onyx object is in the format we expect
+    _.each(data, ({onyxMethod, key, value}) => {
+        if (!_.contains(['set', 'merge'], onyxMethod)) {
+            throw new Error(`Invalid onyxMethod ${onyxMethod} in Onyx update.`);
+        }
+        if (!_.isString(key)) {
+            throw new Error(`Invalid ${typeof key} key provided in Onyx update. Onyx key must be of type string.`);
+        }
+        if (typeof value === 'undefined') {
+            throw new Error(`Undefined value passed to key ${key} in Onyx update.`);
+        }
+    });
+
+    _.each(data, ({onyxMethod, key, value}) => {
+        if (onyxMethod === 'set') {
+            set(key, value);
+            return;
+        }
+        if (onyxMethod === 'merge') {
+            merge(key, value);
+        }
```

Comment:
NAB: I think doing a `switch()` is slightly better here, and it makes it a bit more extensible for the future.

Reply:
Sounds good. I switch (pun intended)